jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/VirtualThreadPool.java

```diff
@@ -0,0 +1,248 @@
+        _threads = tracking ? Collections.newSetFromMap(new WeakHashMap<>()) : null;
+    }
+
+    public boolean isTracking()
+    {
+        return _threads != null;
+    }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        _main = new Thread("virtual main")
+        {
+            @Override
+            public void run()
+            {
+                try (AutoLock.WithCondition l = _joinLock.lock())
+                {
+                    while (isRunning())
+                    {
+                        l.await();
+                    }
+                }
+                catch (InterruptedException e)
+                {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        _main.start();
+
+        _virtualExecutor = Objects.requireNonNull(StringUtil.isBlank(_name)
+            ? VirtualThreads.getDefaultVirtualThreadFactoryExecutor()
+            : VirtualThreads.getNamedVirtualThreadFactoryExecutor(_name));
+        if (_threads != null)
+            _virtualExecutor = new TrackingVirtualExecutor(_virtualExecutor);
+        super.doStart();
+    }
+
+    @Override
+    protected void doStop() throws Exception
+    {
+        super.doStop();
+        _virtualExecutor = null;
+        _main = null;
+
+        try (AutoLock.WithCondition l = _joinLock.lock())
+        {
+            l.signalAll();
+        }
+    }
+
+    @Override
+    public Thread newThread(Runnable r)
+    {
+        return _virtualExecutor.newThread(r);
+    }
+
+    @Override
+    public Executor getVirtualThreadsExecutor()
+    {
+        return _virtualExecutor;
+    }
+
+    @Override
+    public void setVirtualThreadsExecutor(Executor executor)
+    {
+        throw new UnsupportedOperationException("cannot set VirtualThreadExecutor");
+    }
+
+    @Override
+    public void join() throws InterruptedException
+    {
+        try (AutoLock.WithCondition l = _joinLock.lock())
+        {
+            while (isRunning())
+            {
+                l.await();
+            }
+        }
+
+        while (isStopping())
+        {
+            Thread.onSpinWait();
+        }
+    }
+
+    @Override
+    public int getThreads()
+    {
+        return -1;
+    }
+
+    @Override
+    public int getIdleThreads()
+    {
+        return -1;
+    }
+
+    @Override
+    public boolean isLowOnThreads()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean tryExecute(Runnable task)
+    {
+        try
+        {
+            _virtualExecutor.execute(task);
+            return true;
+        }
+        catch (RejectedExecutionException e)
+        {
+            LOG.warn("tryExecute {} failed", _name, e);
+        }
+        return false;
+    }
+
+    @Override
+    public void execute(Runnable task)
+    {
+        _virtualExecutor.execute(task);
+    }
+
+    private class TrackingVirtualExecutor implements VirtualThreads.ThreadFactoryExecutor
```

Comment:
Should be factored out for reusability, move `_threads` into this class, and add dump capabilities.

Reply:
I think we should leave it here until there is a need to reuse.
I have moved _threads and done other cleanups.